### DISCUSSION TITLE
actions.moveBy gets Actor stuck on fixed collision objects

### DISCUSSION THREAD

**Discussion by DaVince** [excaliburjs/Excalibur#2510] [Q&A] [answered]
Hi,

I've been using actions to move my character tile by tile in an animated fashion. It's looked nice so far, but now I'm taking the next step and adding fixed collision objects to the game and the player actor always gets irrevocably stuck as soon as it touches the object.

A reproduction can be found here: https://github.com/DaVince/excalibur-bugs/tree/action-gets-object-stuck/src

The sword gets stuck as soon as it walks into the red box object.

Unfortunately, practically speaking, this means I cannot use a solid Tiled layer and Actions for character movement at the same time...

**Comment by AWaselnuk** [comment, marked as answer]
I'm 100% not sure if this is a good idea (I'm new to game dev + Excalibur) but during prototyping I encountered the same issue because I was just using mouse clicks to have a player `moveTo` things.

The code that fixed it for me was this:

```js
  // clear actions when player collides with something
  player.on("postcollision", (evt) => {
    player.actions.clearActions();
  });
```

Of course this approach made my movement feel pretty rough, but if you are building a gridded, tile-by-tile system this might work fine?

**Reply by DaVince** [reply]
Thanks! That seems to solve the issue of the character freezing up.

I'm now dealing with the character going off the grid instead, but that might be an unrelated issue due to how I'm handling key repeats. At least the player can't get stuck now. :)

**Reply by eonarheim** [reply]
What are you running into with the keys? Are too many events firing than you expect?

**Reply by DaVince** [reply]
No, the amount of events seemed fine. In the end it was something in my collision handling that made the actual position deviate from the grid over time. I solved it by basically force resetting the player position to the middle of the tile after each move.

**Reply by DaVince** [reply]
I ultimately got things working (though it's probably hackier than it could be) so I'm marking this as resolved :slightly_smiling_face:
